tests/backup/backup_restore_basic_test.go

```diff
@@ -2800,3 +2802,410 @@ var _ = Describe("{BackupRestoreOnDifferentK8sVersions}", func() {
+
+					// Validation code will be added here in upcoming PR
+
+					// If Later Restore was an error before, we have to fail the test at this point, having processed the other stage
+					dash.VerifyFatal(restoreLaterStatusErr, nil, fmt.Sprintf("status of later restore [%s] is 'PartialSuccess' or 'Success'", restoreLaterName))
+
+					// Checking actual validity of restore status
+					log.InfoD("Validating status of Initial and Later Restore...")
+					validity := false
+					errHelpStr := ""
+					log.InfoD("states of [initial,later] restore are [%s,%s]", initialRestoreStatus, laterRestoreStatus)
+					if (initialRestoreStatus == api.RestoreInfo_StatusInfo_Success && laterRestoreStatus == api.RestoreInfo_StatusInfo_PartialSuccess) ||
+						(initialRestoreStatus == api.RestoreInfo_StatusInfo_PartialSuccess && laterRestoreStatus == api.RestoreInfo_StatusInfo_Success) {
+						validity = true
+					} else if initialRestoreStatus == api.RestoreInfo_StatusInfo_PartialSuccess && laterRestoreStatus == api.RestoreInfo_StatusInfo_PartialSuccess {
+						validity = false
+						errHelpStr = "Error. help: ensure no remnant cluster-level resources on destination cluster."
+					} else if initialRestoreStatus == api.RestoreInfo_StatusInfo_Success && laterRestoreStatus == api.RestoreInfo_StatusInfo_Success {
+						validity = false
+						errHelpStr = "Error. help: ensure app has cluster-level resources."
+					}
+					dash.VerifyFatal(validity, true, fmt.Sprintf("states of (initial,later) restore are (Success,PartialSuccess) or (PartialSuccess,Success).\n %s", errHelpStr))
+				})
+
+			}
+		})
+
+	})
+
+	JustAfterEach(func() {
+		defer func() { Inst().AppList = originalAppList }()
+
+		log.InfoD("Begin JustAfterEach")
+		defer func() { log.InfoD("End JustAfterEach") }()
+
+		defer func() {
+			log.InfoD("switching to default context")
+			err := SetClusterContext("")
+			log.FailOnError(err, "failed to SetClusterContext to default cluster")
+		}()
+
+		defer EndTorpedoTest()
+
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "fetching px-central-admin ctx")
+
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = false
+		log.InfoD("deleting deployed applications for source and destination clusters")
+
+		log.InfoD("switching to source context")
+		err = SetSourceKubeConfig()
+		log.FailOnError(err, "failed to switch to context to source cluster")
+
+		log.InfoD("deleting deployed applications on source clusters")
+		ValidateAndDestroy(scheduledAppContexts, opts)
+
+		log.InfoD("waiting (for 1 minute) for any Resources created by Operator of Custom Resources to finish being destroyed.")
+		time.Sleep(time.Minute * 1)
+		log.Warn("no verification of destruction is done; it might lead to undetectable errors")
+
+		// Restored stuff destruction will be added here in upcoming PR
```

Comment:
You need this comment?

Reply:
it probably won't be added at this point, based on my discussion with Mithun that this will be run on clusters which will be purged after the test is run, thus not requiring clean up of resources.

✅ removed comment